components/command/filemanager.go

```diff
@@ -32,8 +32,12 @@ func (fm *FileManager) CreateDirectoryFromPulumiString(name string, remotePath p
 }
 
 // CreateDirectory if it does not exist
+// To avoid pulumi.URN colisions if multiple files use the same directory, use the full filePath as URN and path.Split out the folderPath for creation
 func (fm *FileManager) CreateDirectory(remotePath string, useSudo bool, opts ...pulumi.ResourceOption) (*remote.Command, error) {
-	return fm.command.CreateDirectory(fm.runner, "create-directory-"+remotePath, pulumi.String(remotePath), useSudo, opts...)
+	// if given just a directory path, path.Split returns "" as file
+	//  eg. path.Split("/a/b/c/") -> "/a/b/c/", ""
+	folderPath, _ := path.Split(remotePath)
+	return fm.command.CreateDirectory(fm.runner, "create-directory-"+remotePath, pulumi.String(folderPath), useSudo, opts...)
```

Comment:
```suggestion
	// if given just a directory path, path.Split returns "" as file
	//  eg. path.Split("/a/b/c/") -> "/a/b/c/", ""
	folderPath, _ := path.Split(remotePath)
	return fm.command.CreateDirectory(fm.runner, "create-directory-"+remotePath, pulumi.String(folderPath), useSudo, opts...)
}

// CreateDirectoryForFile if it does not exist
func (fm *FileManager) CreateDirectoryForFile(fullFilePath string, useSudo bool, opts ...pulumi.ResourceOption) (*remote.Command, error) {
	remotePath, _ := path.Split(fullFilePath)
	return fm.command.CreateDirectory(fm.runner, "create-directory-"+fullFilePath, pulumi.String(remotePath), useSudo, opts...)
```

Reply:
fixed in https://github.com/DataDog/test-infra-definitions/pull/399/commits/74520197adec21523407119c74cfd3e0c7394b24